### ISSUE TITLE
Pooling doesn't work?

### ISSUE THREAD

**Issue by nugzta** [asgardex/asgard-exchange#70]
I tried to test the pooling on the testnet. I swapped 500 tRUNE into 0.01900707 tBTC. Then I added liquidity, it took my tRUNE but not tBTC. Now, I end up with only 0.98 tRUNE and still 0.01900707 tBTC in my wallets.

These are my wallets:

https://testnet-explorer.binance.org/address/tbnb1pvsjrdef7zvher0clg95am5k2q5acnydz8teah

https://blockstream.info/testnet/address/tb1qss5zvwp0w7k4nq9tten3mrqpny2unf6gdgqa27

In the transactions, it clearly shows tRUNE was taken but not tBTC.

**Comment by stuartkuentzel** [comment]
Hey @nugzta, did the tBTC transaction eventually go through? 

One of the issues we're hitting with the UI is that BNB processes very quickly, but it takes a little more time for THORChain to process, and then more time for BTC. The user may think the transaction is done because they see BNB has gone through, but the THORChain / BTC txs are still pending. This is a UI problem, we're working on a better flow for it.

**Comment by nugzta** [comment]
> Hey @nugzta, did the tBTC transaction eventually go through?
> 
> One of the issues we're hitting with the UI is that BNB processes very quickly, but it takes a little more time for THORChain to process, and then more time for BTC. The user may think the transaction is done because they see BNB has gone through, but the THORChain / BTC txs are still pending. This is a UI problem, we're working on a better flow for it.

Nope, I still have 0.98 tRUNE and still 0.01900707 tBTC in my wallets. So probably just as you said, when I added the liquidity, it took my tRUNE but my tBTC was still transferring to my wallet, so now I lost my tRUNE.

I tried to add liquidity again: https://tinyurl.com/y6xvslao

tRUNE is gone, insufficient balance to provide liquidity.

**Comment by nugzta** [comment]
Just to be clear, the swap was fine. I started with 1000 tRUNE, I swapped about 500 tRUNE and received 0.01900707 tBTC. After that, the pooling failed. It took tRUNE and I'm only left with 0.01900707 tBTC.

**Comment by gchatrian** [comment]
Hi,

I tried Asgard yesterday and I had the same issues:

- Swapped RUNE for BTC: OK
- Pooled BTC+RUNE: KO

I don't know where the assets went. No transaction recorded, non liquidity pool provided.

**Comment by stuartkuentzel** [comment]
@nugzta just to clarify, your balance was 500 tRUNE and 0.01900707 tBTC. You're saying you tried and pool 100% of that balance? And it pooled only the 500 tRUNE but didn't take any of the BTC?

**Comment by stuartkuentzel** [comment]
@gchatrian just tested pooling BTC + RUNE and it processed. There are some UI changes that need to be made to poll the pool list and refresh automatically on changes. If you check again now, are your funds listed?

**Comment by nugzta** [comment]
> @nugzta just to clarify, your balance was 500 tRUNE and 0.01900707 tBTC. You're saying you tried and pool 100% of that balance? And it pooled only the 500 tRUNE but didn't take any of the BTC?

Yup, I had 500 tRUNE and 0.01900707 tBTC. I wanted to pool all of them. It took 500 tRUNE while probably tBTC was still in the process of swapping/transferring between wallets. Now I only have 0.01900707 tBTC. Nothing is pooled -> No deposited pools found.

https://tinyurl.com/y3apkmgo

**Comment by gchatrian** [comment]
> 
> 
> @gchatrian just tested pooling BTC + RUNE and it processed. There are some UI changes that need to be made to poll the pool list and refresh automatically on changes. If you check again now, are your funds listed?

HI there,

I still see "no deposited pools found". It was 1000 RUNE + 0.041 BTC. Wallet is asgardex-tbnb15p_u86

**Comment by stuartkuentzel** [comment]
@gchatrian @nugzta what are the BNB addresses and BTC addresses associated with your keystore?

**Comment by gchatrian** [comment]
HI,

BTC: tb1qnk9pagmftake9eq4w4rgm3typnlwf8zfusqcsj
BNB: tbnb15p9nxwudh8dg8034e6ftg682u0zafhg2229u86

Thanks,
Giulio

**Comment by stuartkuentzel** [comment]
@nugzta @gchatrian thanks for reporting this. it looks like an uncaught error we need to fix. in both cases, the RUNE transactions were successful, but the Bitcoin transaction failed. so the deposit is technically still pending on THORChain, waiting for bitcoin. 

would you mind trying another deposit? if it fails again, it'll be helpful cause we'll know something is up with the bitcoin client for you both. if it succeeds, my guess is that your initial RUNE deposit + the new deposit will both show up, and you'll be able to withdraw everything.

**Comment by gchatrian** [comment]
Hi,

I just swapped 500 RUNE for 0.0XX (sry i did not write down the exact amt) BTC. Everything went ok.
When added 408 RUNE + 0.0x BTC to the liquidity pool i got a message the TX was ok BUT 
https://blockstream.info/testnet/tx/c6f95d10ee5e3d86fd7fecf548afec4ba5edba03a79959a403096395c3170a19 says it could not find the transaction and no liquidity is showing "no deposited pools found". Now I am out of funds

**Comment by stuartkuentzel** [comment]
@gchatrian it looks like THORChain is recognizing the stake. It takes some time for bitcoin transactions to process. On blockstream now, that transaction is recognized as well.

**Comment by nugzta** [comment]
> @nugzta @gchatrian thanks for reporting this. it looks like an uncaught error we need to fix. in both cases, the RUNE transactions were successful, but the Bitcoin transaction failed. so the deposit is technically still pending on THORChain, waiting for bitcoin.
> 
> would you mind trying another deposit? if it fails again, it'll be helpful cause we'll know something is up with the bitcoin client for you both. if it succeeds, my guess is that your initial RUNE deposit + the new deposit will both show up, and you'll be able to withdraw everything.

How do I try another deposit if I don't have any tRUNE?

**Comment by stuartkuentzel** [comment]
@nugzta created an issue with the midgard API to display a user pool if, for whatever reason, only one half is deposited. that way the UI can recognize the user pool and you can interact with it.

https://gitlab.com/thorchain/midgard/-/issues/319

**Comment by gchatrian** [comment]
Hi,

I can now see the liquidity I added. It just took a lot of time

**Comment by stuartkuentzel** [comment]
Closing this, as it's a bit outdated and there have been a ton of updates including changing our BTC API provider from Blockchair to Sochain + BlockStream, plus how we track pending transactions. If it occurs again, feel free to reopen or submit a new issue.
